src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -161,22 +190,46 @@
                                :swap-proposal          (when-not (empty? best-routes)
                                                          (assoc (first best-routes) :uuid request-uuid))
                                :error-response         error-response
-                               :loading-swap-proposal? false)}
+                               :loading-swap-proposal? false
+                               :initial-response?      false)}
+         (and initial-response? (seq best-routes))
+         (assoc :fx
+                [[:dispatch
+                  [:centralized-metrics/track :metric/swap-proposal-received
+                   {:receive_token  (:symbol asset-to-receive)
+                    :receive_amount (-> amount-out
+                                        (number/hex->whole receive-token-decimals)
+                                        (money/to-fixed receive-token-decimals))}]]])
+         (and initial-response? (empty? best-routes))
+         (assoc :fx
+                [[:dispatch
+                  [:centralized-metrics/track :metric/swap-proposal-failed
+                   {:error (:details error-response)}]]])
```

Comment:
Is this safe ? Can this expose the user's wallet address ?

Reply:
Answered here: https://github.com/status-im/status-mobile/pull/21485#issuecomment-2430806362